festival/src/main/java/likelion/festival/notification/service/NotificationService.java

```diff
@@ -0,0 +1,73 @@
+package likelion.festival.notification.service;
+
+import likelion.festival.notification.dto.NotificationRequestDto;
+import likelion.festival.notification.dto.NotificationResponseDto;
+import likelion.festival.notification.entity.Notification;
+import likelion.festival.notification.entity.NotificationType;
+import likelion.festival.global.exception.WrongNotificationId;
+import likelion.festival.notification.repository.NotificationRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class NotificationService {
+
+  private final NotificationRepository notificationRepository;
+
+  public NotificationResponseDto readNotification(Long id) {
+    Notification notification = notificationRepository.findById(id)
+        .orElseThrow(() -> new WrongNotificationId());
+
+    NotificationResponseDto notificationResponseDto = NotificationResponseDto.toDto(notification);
+
+    return notificationResponseDto;
+  }
+
+  public List<NotificationResponseDto> readNotificationAll(NotificationType notificationType) {
+    List<NotificationResponseDto> notificationResponseDtos = new ArrayList<>();
+    if (notificationType == null) {
+      List<Notification> notifications = notificationRepository.findAll();
+      for (Notification notification : notifications) {
+        NotificationResponseDto notificationResponseDto = NotificationResponseDto.toDto(notification);
+        notificationResponseDtos.add(notificationResponseDto);
+      }
+      return notificationResponseDtos;
+    }
+    List<Notification> notifications = notificationRepository.findByNotificationType(
+        notificationType);
+    for (Notification notification : notifications) {
+      NotificationResponseDto notificationResponseDto = NotificationResponseDto.toDto(notification);
+      notificationResponseDtos.add(notificationResponseDto);
+    }
+    return notificationResponseDtos;
+  }
+
+  @Transactional
+  public Notification createNotification(NotificationRequestDto notificationRequestDto) {
+    Notification notification = notificationRequestDto.toEntity(notificationRequestDto);
+    return notificationRepository.save(notification);
+  }
+
+  @Transactional
+  public void deleteNotification(Long id) {
+    notificationRepository.deleteById(id);
+  }
+
+  @Transactional
+  public Notification updateNotification(Long id, NotificationRequestDto notificationRequestDto) {
+    Notification notification = notificationRepository.findById(id)
+        .orElseThrow(() -> new WrongNotificationId());
+
+
+    notification.updateNotification(notificationRequestDto.getTitle(), notificationRequestDto.getWriter(),
+        notificationRequestDto.getContent(), notificationRequestDto.getNotificationType(),
+        notificationRequestDto.getImages());
```

Comment:
Notificaion의 상태(변수)가 변경될 때 마다 해당 부분에 대한 코드 수정이 필요해보이는데요~
Notificaion의 변경에 영향을 받지 않도록 코드를 구현하면 어떨까요?

```suggestion
    Notification newNotification = notificationRequestDto.toEntity();
    notification.updateNotification(newNotification);
```